hazelcast/src/main/java/com/hazelcast/internal/server/ServerConnectionManager.java

```diff
@@ -76,11 +78,19 @@ default boolean register(Address remoteAddress, ServerConnection connection) {
      * the relevant {@link Connection} resource.
      *
      * @param remoteAddress - The remote address to register the connection under
+     * @param targetAddress - Requested target address on the connector side, null if acceptor side
      * @param connection    - The connection to be registered
      * @param planeIndex    - The index of the plane
      * @return True if the call was successful
      */
-    boolean register(Address remoteAddress, ServerConnection connection, int planeIndex);
+    boolean register(
+            Address remoteAddress,
+            Address targetAddress,
```

Comment:
I changed the name of this remote address to primary address and added a comment on it. In general, I tried to use MEMBER endpoint's public address, if it exists, obtained from `MemberHandshake` as primary address. If the MEMBER public address is not provided, this can happen in advanced network. In this case, I use the available public address of the related protocol as the primary address and register the connection under it. 
These public addresses may differ from the target address of the connection initiator side.